test/function/expression_body.unit

```diff
@@ -0,0 +1,71 @@
+40 columns                              |
+### Tests for named function declarations whose body is `=>`.
+>>> Unsplit body.
+f  (  )  =>  body  ;
+<<<
+f() => body;
+>>> Async.
+f  (  )  async  =>  body  ;
+<<<
+f() async => body;
+>>> Sync*.
+f  (  )  sync  *  =>  body  ;
+<<<
+f() sync* => body;
+>>> Async*.
+f  (  )  async  *  =>  body  ;
+<<<
+f() async* => body;
+>>> Split at `=>`.
+myFunction() => argument + anotherArgument;
+<<<
+myFunction() =>
+    argument + anotherArgument;
+>>> Split inside expression forces split at `=>`.
+veryLongFunction() => argument + anotherArgument + aThirdArgument;
+<<<
+veryLongFunction() =>
+    argument +
+        anotherArgument +
```

Comment:
It's a little odd how these don't line up properly.

Reply:
Yeah, that's a fair observation. The old style [handles that case explicitly](https://github.com/dart-lang/dart_style/blob/main/lib/src/source_visitor.dart#L344-L352), at least for the RHS of expression-bodied functions. I'm up for doing that in the new style too. Added a TODO to AssignPiece (which uniformly handles `=>`, `=`, and `:` in the new style).